pkg/segment/context/context.go

```diff
@@ -0,0 +1,61 @@
+package context
+
+import (
+	"context"
+	"sync"
+)
+
+type contextKey struct{}
+
+var key = contextKey{}
+
+type properties struct {
+	lock    sync.Mutex
+	storage map[string]interface{}
+}
+
+func (p *properties) set(name string, value interface{}) {
+	p.lock.Lock()
+	defer p.lock.Unlock()
+	p.storage[name] = value
+}
+
+func (p *properties) values() map[string]interface{} {
+	p.lock.Lock()
+	defer p.lock.Unlock()
+	ret := make(map[string]interface{})
+	for k, v := range p.storage {
+		ret[k] = v
```

Comment:
You are doing this to return a deep copy of the map, right?
Are we sure all values of the map will be copied this way? 
Here, If v were a map, I think it would be referenced instead

Reply:
Yes, for a deep copy. The chances of v being a map are pretty slim in the case of telemetry, but I guess we can use json instead to fix this.

Reply:
To be sure that we have only strings (and ints probably later) stored in the map, you can set as private the `setContextProperties` (as in the original source), and place `SetComponentType` in the segment/context package, so only this method is callable and is allowed to set values into the map, so we are sure values are strings, which will be copied correctly